File: lua/config/options.lua
-- luacheck: globals vim

if vim.env.DISPLAY ~= nil then
	-- use system clipboard
	-- https://vim.fandom.com/wiki/Accessing_the_system_clipboard
	vim.o.clipboard="unnamedplus"
end


vim.opt.foldlevel = 99

vim.opt.encoding = "utf8"
vim.opt.mouse = "nv" --"a"
vim.opt.backspace = "eol,start,indent"
vim.opt.wrap = true

vim.opt.errorbells = false
vim.opt.visualbell = false

vim.opt.number = true
vim.opt.relativenumber = true
vim.opt.list = true
vim.opt.listchars = {
	tab = '› ',
	lead = '·',
	trail = '·',
}
vim.o.background = "dark"

vim.opt.incsearch = true
vim.opt.hlsearch = true
vim.opt.ignorecase = true
vim.opt.smartcase = true

-- $l -> next line char 0
--vim.opt.whichwrap:append({'<', '>', '[', ']'})
vim.cmd([[
	set whichwrap+=<,>,[,]
]])

File: src/config/options.c
#include "../nvim.h"

int luaopen_config_options() {
  o("foldlevel", i(99));
  o("encoding", s("utf8"));

  // design
  o("list", True);
  o("wrap", True);
  o("errorbells", False);
  o("visualbell", False);
  o("background", s("dark"));
  o("number", True);
  o("relativenumber", True);
  //o("numberwidth", i(2));

  // search
  o("incsearch", True);
  o("hlsearch", True);
  o("ignorecase", True);
  o("smartcase", True);

  // Controls
  o("mouse", s("nv"));
  o("whichwrap", s("<>[]hlb,s"));
  o("backspace", s("eol,start,indent"));

  // Filetypes
  g("python_recommended_style", i(0));

  //Dictionary listkeys = ARRAY_DICT_INIT;
  //PUT(listkeys, "tab", s("› "));
  //PUT(listkeys, "lead", s(" "));
  //PUT(listkeys, "trail", s("·"));
  //o("listchars", DICTIONARY_OBJ(listkeys));

  return 1;
}
